catalog/spatial/geocoding/spatial-geocoding-plugin/src/main/java/org/codice/ddf/spatial/geocoding/plugin/GeoCoderPlugin.java

```diff
@@ -0,0 +1,126 @@
+import ddf.catalog.operation.CreateRequest;
+import ddf.catalog.operation.DeleteRequest;
+import ddf.catalog.operation.UpdateRequest;
+import ddf.catalog.plugin.PluginExecutionException;
+import ddf.catalog.plugin.PreIngestPlugin;
+import ddf.catalog.plugin.StopProcessingException;
+import ddf.catalog.util.impl.ServiceSelector;
+
+/**
+ * The {@code GeoCoderPlugin} is responsible for adding a {@link Location#COUNTRY_CODE} in ISO 3166-1 format
+ * to {@code Metacard}s that have the {@link Metacard#GEOGRAPHY} attribute.
+ */
+public class GeoCoderPlugin implements PreIngestPlugin {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(GeoCoderPlugin.class);
+
+    private static ServiceSelector<GeoCoder> geoCoderFactory;
+
+    private int radiusInKm = 10;
+
+    public GeoCoderPlugin(ServiceSelector<GeoCoder> geoCoderFactory) {
+        if (geoCoderFactory == null) {
+            throw new IllegalArgumentException(
+                    "GeoCoderPlugin(): constructor argument 'geoCoderFactory' may not be null.");
+        }
+
+        this.geoCoderFactory = geoCoderFactory;
+    }
+
+    @Override
+    public CreateRequest process(CreateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        if (input.getMetacards() == null) {
+            return input;
+        }
+
+        for (Metacard metacard : input.getMetacards()) {
+            Optional<String> wktLocation = Optional.ofNullable(metacard.getLocation());
+
+            if (wktLocation.isPresent()) {
+                if (countryCodeNeeded(metacard)) {
+                    GeoCoder geocoder = geoCoderFactory.getService();
+
+                    if (geocoder != null) {
+                        Optional<String> alpha3CountryCode =
+                                geocoder.getCountryCode(wktLocation.get(), radiusInKm);
+
+                        if (alpha3CountryCode.isPresent()) {
+                            setCountryCodeAttribute(metacard, alpha3CountryCode.get());
+                            LOGGER.trace(
+                                    "Setting metacard country code to {} for metacard with id {}",
+                                    alpha3CountryCode.get(),
+                                    metacard.getId());
+                        }
+                    }
+                }
+            }
+        }
+        return input;
+    }
+
+    @Override
+    public UpdateRequest process(UpdateRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        // TODO: Implement when AdaptedMetacard updated to use taxonomy
+        return input;
+    }
+
+    @Override
+    public DeleteRequest process(DeleteRequest input)
+            throws PluginExecutionException, StopProcessingException {
+        return input;
+    }
+
+    public void updateConfiguration(Map<String, Object> properties) {
+        LOGGER.trace("Updating GeoCoderPlugin search radius");
+
+        if (properties != null) {
```

Comment:
I think you could do something like: Optional.of(properties).map(p -> { return p.get("radiusInKm");}.filter(Integer.class::isInstance).map(Integer.class::cast).ifPresent(value ->{ setRadiusInKm(value); }); 

That way, you lower the branch complexity.